src/main/java/cuchaz/enigma/analysis/index/PackageVisibilityIndex.java

```diff
@@ -0,0 +1,144 @@
+package cuchaz.enigma.analysis.index;
+
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+import cuchaz.enigma.analysis.EntryReference;
+import cuchaz.enigma.translation.representation.AccessFlags;
+import cuchaz.enigma.translation.representation.entry.*;
+
+import java.util.*;
+
+public class PackageVisibilityIndex implements JarIndexer {
+	private static boolean isPackageVisibleOnlyRef(AccessFlags entryAcc, EntryReference ref, InheritanceIndex inheritanceIndex) {
+		if (entryAcc.isPublic()) return false;
+		if (entryAcc.isProtected()) {

```

Comment:
Hmm, would package-private be relevant here?

Reply:
Package private just returns true here. This logic appears good to me.

Reply:
Ah, right. Misread it.